app/models/concerns/terrier/script_base.rb

```diff
@@ -71,6 +71,12 @@ def compute_field_options
     @visibility_levels = %w(public private)
     enum_field :visibility, @visibility_levels
 
+    ## Email
+
+    def send_email_if_necessary(output_files = [], log_url = "", user_full_name = "")
+      ""
+    end
```

Comment:
It doesn't seem like this is actually complete. Should be something like:
```ruby
if self.email_recipients.present?
  begin
    # this needs to be implemented by the clypboard and lycensed Script classes:
    send_email output_files, log_url, user
  rescue => ex
    Rails.logger.warn "Error sending script e-mail: #{ex.message}"
  end
end
```

Also, we should continue to pass around the full user object, not just the name, so that the action log will contain the user's id as well.